pkg/ingester/ingester.go

```diff
@@ -407,7 +415,17 @@ func (i *Ingester) starting(ctx context.Context) error {
 		servs = append(servs, closeIdleService)
 	}
 
-	var err error
+	marker, err := i.shutdownMarker.Exists()
+	if err != nil {
+		return errors.Wrap(err, "failed to check ingester shutdown marker")
+	}
+
+	if marker {
+		level.Info(i.logger).Log("msg", "detected existing shutdown marker, setting unregister and flush on shutdown", "path", i.shutdownMarker.Path)
+		i.lifecycler.SetUnregisterOnShutdown(true)
+		i.lifecycler.SetFlushOnShutdown(true)
```

Comment:
Should these two calls and the corresponding calls in `PrepareShutdownHandler` be extracted to a single place so that we always do the same thing in both situations?